### PR TITLE
Bump to latest kubevirtci that includes ocp-4.3

### PR DESCRIPTION
**Pull request by oshoval**
Signed-off-by: Or Shoval <oshoval@redhat.com>

```release-note
NONE
```

### CODE REVIEW

**Comment by alonSadan** [comment]
looks good to me

**Comment by rmohr** [comment]
`make cluster-sync` fails with:

```
no ssh port found for node01
```

I guess the node names need to be adjusted.

**Comment by rmohr** [comment]
/hold

Can you please ensoure that there is an optional prowjob for it, so that we can see it running?

**Comment by oshoval** [comment]
can i fix the problem (already created a fix) on a new PR ?
since it isnt because of the bump and not connected

and will add the lane next week
it better to be optional imo indeed and not blocker (not blocking at least at the beginning)

thanks

**Comment by rmohr** [comment]
Why can't we just prepare everything until it works and then merge it when done? What is the benefit of the incremental merges?

**Comment by oshoval** [comment]
people already using ocp-4.3 manually
for example now with the ssh.sh fixed they will also be able to use make sync

there are more issues that we need to prioritze
and if i add a lane, there are little issues here and there that we fix manually (for example hanged pod that can be deleted manually in order to cluster to finish), and if we block this PR due to the lane, it will take time and people wont able to proceed on stuff they need imo

**Comment by rmohr** [comment]
> people already using ocp-4.3 manually
> for example now with the ssh.sh fixed they will also be able to use make sync
> 

They seem to not use it withing kubevirt/kubevirt since it did not work.

> there are more issues that we need to prioritze
> and if i add a lane, there are little issues here and there that we fix manually (for example hanged pod that can be deleted manually in order to cluster to finish), and if we block this PR due to the lane, it will take time and people wont able to proceed on stuff they need imo

Talking about introducing ocp-4.3 in a limited fashion, like with not running all tests is ok. Introducing it as basically dead code is not.

**Comment by oshoval** [comment]
once 
https://github.com/kubevirt/kubevirtci/pull/225
will be merged, need to rebump

**Comment by oshoval** [comment]
rebumped to latest needed kubevirtci changes

**Comment by dhiller** [comment]
Test failure for ceph lane looks unrelated.

/override pull-kubevirt-e2e-k8s-1.15.1-ceph

**Comment by oshoval** [comment]
@rmohr 
can we unhold it ?
since you created that prow job, and all passes ?
thanks

**Comment by dhiller** [comment]
@oshoval I have just started the optional test lane to see how it goes. I don't think that this will block the merge, just making sure everything works. We can then fix issues in a later PR. @rmohr agreed?

**Comment by oshoval** [comment]
ocp fails on the end of make cluster-build

if i ran locally i get the next lines, so maybe it doesnt have access to push ? (well it shouldnt anyway i guess, unless its locally)
```
Step 7/7 : ENTRYPOINT [ "/entrypoint.sh" ]
 ---> Using cache
 ---> d04363484108
Successfully built d04363484108
Successfully tagged kubevirt:latest
go version go1.12.8 linux/amd64

go version go1.12.8 linux/amd64
INFO: Writing tracer profile to '/root/.cache/bazel/_bazel_root/6f347497f91c9a385dcd9294645b76e0/command.profile.gz'
INFO: Build options --define, --platforms, and --stamp have changed, discarding analysis cache.
INFO: Analyzed target //:push-other-images (70 packages loaded, 9723 targets configured).
INFO: Found 1 target...
Target //:push-other-images up-to-date:
  bazel-bin/push-other-images
INFO: Elapsed time: 39.269s, Critical Path: 35.36s
INFO: 0 processes.
INFO: Build completed successfully, 15 total actions
INFO: Build completed successfully, 15 total actions
2020/01/09 10:57:51 Error pushing image to kubevirt/cdi-http-import-server:devel: unable to push image to kubevirt/cdi-http-import-server:devel: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:kubevirt/cdi-http-import-server Type:repository] map[Action:push Class: Name:kubevirt/cdi-http-import-server Type:repository]]
2020/01/09 10:57:52 Error pushing image to kubevirt/fedora30-cloud-container-disk-demo:devel: unable to push image to kubevirt/fedora30-cloud-container-disk-demo:devel: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:kubevirt/fedora30-cloud-container-disk-demo Type:repository] map[Action:push Class: Name:kubevirt/fedora30-cloud-container-disk-demo Type:repository]]
2020/01/09 10:57:52 Error pushing image to kubevirt/cirros-container-disk-demo:devel: unable to push image to kubevirt/cirros-container-disk-demo:devel: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:kubevirt/cirros-container-disk-demo Type:repository] map[Action:push Class: Name:kubevirt/cirros-container-disk-demo Type:repository]]
2020/01/09 10:57:52 Error pushing image to kubevirt/disks-images-provider:devel: unable to push image to kubevirt/disks-images-provider:devel: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:kubevirt/disks-images-provider Type:repository] map[Action:push Class: Name:kubevirt/disks-images-provider Type:repository]]
2020/01/09 10:57:52 Error pushing image to kubevirt/fedora-cloud-container-disk-demo:devel: unable to push image to kubevirt/fedora-cloud-container-disk-demo:devel: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:kubevirt/fedora-cloud-container-disk-demo Type:repository] map[Action:push Class: Name:kubevirt/fedora-cloud-container-disk-demo Type:repository]]
2020/01/09 10:57:52 Error pushing image to kubevirt/winrmcli:devel: unable to push image to kubevirt/winrmcli:devel: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:kubevirt/winrmcli Type:repository] map[Action:push Class: Name:kubevirt/winrmcli Type:repository]]
2020/01/09 10:57:52 Error pushing image to kubevirt/cirros-custom-container-disk-demo:devel: unable to push image to kubevirt/cirros-custom-container-disk-demo:devel: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:kubevirt/cirros-custom-container-disk-demo Type:repository] map[Action:push Class: Name:kubevirt/cirros-custom-container-disk-demo Type:repository]]
2020/01/09 10:57:52 Error pushing image to kubevirt/virtio-container-disk:devel: unable to push image to kubevirt/virtio-container-disk:devel: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:kubevirt/virtio-container-disk Type:repository] map[Action:push Class: Name:kubevirt/virtio-container-disk Type:repository]]
2020/01/09 10:57:52 Error pushing image to kubevirt/example-hook-sidecar:devel: unable to push image to kubevirt/example-hook-sidecar:devel: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:kubevirt/example-hook-sidecar Type:repository] map[Action:push Class: Name:kubevirt/example-hook-sidecar Type:repository]]
make: *** [Makefile:107: cluster-build] Error 1
[root@modi01 kubevirt]# make cluster-build
```

**Comment by oshoval** [comment]
btw 
in the prow log i see stuff like

```
+ WINDOWS_LOCK_PATH=/var/lib/stdci/shared/download_windows_image.lock
+ [[ ocp-4.3 =~ os-.* ]]
+ [[ ocp-4.3 =~ okd-.* ]]
+ [[ ocp-4.3 =~ windows.* ]]
+ export NAMESPACE=kubevirt
+ NAMESPACE=kubevirt
```

maybe there are places we should add ocp as legit name for some commands that search patterns ?

**Comment by rmohr** [comment]
that looks like it does not pick up the name of the local registry. Instead it tries to push to docker hub.

**Comment by qinqon** [comment]
Can we change https://github.com/kubevirt/kubevirt/blob/master/hack/cluster-build.sh#L24 to get `-xe` ?

**Comment by oshoval** [comment]
ocp lane is broken since there is a bug in kubevirtci ssh.sh on ocp (i guess also on okd lanes possible),

haproxy cfg files point to ip for worker that ends with 51
since there is a machineconfig update, which spins another node , let the first one evicted and be upgraded, and then delete one of the workers (if i understood right), 
the ip is sometimes 51 and sometimes 52
so haproxy need to be updated,

run.sh already have part of updating some of the resources according the actual ip
we need to add there an update to the cfg as well, and restart haproxy
new provider should be made

File: automation/test.sh [hunk 1]
```diff
@@ -27,6 +27,7 @@
 #     TODO: vagrant-tagged-release: Runs all possible functional tests on a release deployment in vagrant on a tagged release
 
 set -ex
+set -x
```
Comment (oshoval) [inline automation/test.sh]: i will remove this and the one on the other file, once it works

**Comment by oshoval** [comment]
travis fails because images.sh here was updated manually before it was merged,
rebase would fix it, will do later so first the ocp lane will be finished

**Comment by oshoval** [comment]
/test pull-kubevirt-e2e-k8s-1.15.1-ceph
/test pull-kubevirt-e2e-k8s-1.16.2 
rerunning because log NA

failed to get job history: failed to locate build data: failed to read pr-logs/directory/pull-kubevirt-e2e-k8s-1.15.1-ceph/latest-build.txt: failed to get reader for GCS object: Get https://storage.googleapis.com/kubevirt-prow/pr-logs/directory/pull-kubevirt-e2e-k8s-1.15.1-ceph/latest-build.txt: oauth2: cannot fetch token: Post https://oauth2.googleapis.com/token: dial tcp: lookup oauth2.googleapis.com on 38.145.50.79:53: server misbehaving

**Comment by oshoval** [comment]
/retest
but should we block on this changes even that they dont mess with k8s provider ?
(and also passed before the last trivial rebase, i think)

**Comment by dhiller** [comment]
I have hopes that the changes will go in with tide. Will override the ceph lane if it does not pass.

**Comment by rmohr** [comment]
The ceph lane is very instable the last days. That's new.

**Comment by oshoval** [comment]
what we are going to do about the ceph ?

**Comment by rmohr** [comment]
> what we are going to do about the ceph ?

@oshoval the PR is now fully approved and not in hold state. Therefore you will see from time to time a retest comment added by an extra bot until all tests are green and the PR goes in. The results of the builds will be collected and we will then see together with all other PRs issues collected in the flakefinder report and can then work on removing the cause of the extra failures, if they even exist and it was not a subjective feeling from me that the ceph lane is more flaky. Here are the flakefinder reports: https://storage.googleapis.com/kubevirt-prow/reports/flakefinder/kubevirt/kubevirt/index.html

**Pull request merged**